makefiles/docker.inc.mk

```diff
@@ -82,6 +82,49 @@ DOCKER_OVERRIDE_CMDLINE := $(strip $(DOCKER_OVERRIDE_CMDLINE))
 # Overwrite if you want to use `docker` with sudo
 DOCKER ?= docker
 
+
+# terminating '/' in patsubst is important to match $1 == $(RIOTBASE)
+define dir_is_outside_riotbase
+$(filter $1/,$(patsubst $(RIOTBASE)/%,%,$1/))
+endef
+
+# $1 = riotcpu  $($1) = RIOTCPU  $($($1)) = $(RIOTCPU)
+# -v '$(RIOTCPU):$(DOCKER_BUILD_ROOT)/riotcpu'
+# -e 'RIOTCPU=$(DOCKER_BUILD_ROOT)/riotcpu'
+define docker_volume_and_env
+  -v '$($($(1))):$(DOCKER_BUILD_ROOT)/$1'
+  -e '$1=$(DOCKER_BUILD_ROOT)/$1'
+endef
+
+# Add volumes for possible outside directories
+
+DOCKER_RIOT_DIRS = riotcpu riotboard riotmake riotproject
+# Names with uppercase
```

Comment:
This needs a better comment. I think this is a mapping from directory name inside the docker container to environment variable name for the path outside docker?

Reply:
I agree, a better comment would help. My understanding is:

It is RIOT directories that can be remapped outside of the RIOT repository.
When mounting volumes in docker, if they are not in the RIOT repository, they need to be mounted as well to be accessible and the environment variable set to the mounted volume.

However, I just thought that it does not correctly handle if someone puts a CPU inside his RIOT repository in another directory. Because the environment variable should be set anyway.
I will fix this.

Reply:
I removed this as there are not that many variables and was more confusing than anything.